### ISSUE TITLE
Fail on import of 2.1 (3.0) mesh

### ISSUE THREAD

**Issue by cryham** [OGRECave/blender2ogre#187]
#### System Information
- Ogre Version: 3.0, master
- Operating System / Platform: Debian 12
- Blender Version: 3.6.4

#### Detailed description
I'm trying to import a .mesh file, it has `MeshSerializer_v2.1 R2` inside.
Is this supported? Am I doing something wrong?
File balloon1.mesh here:
https://github.com/stuntrally/stuntrally3/blob/main/data/objects/balloon1.mesh

#### blender2ogre.log
```
2023-11-26 17:07:55 [ INFO] Target_path: /home/ch/_sr/_sr3/data/objects
2023-11-26 17:07:55 [ INFO] Target_file_name: balloon1.mesh
2023-11-26 17:07:55 [ INFO] * Loading mesh from: /home/ch/_sr/_sr3/data/objects/balloon1.mesh
2023-11-26 17:07:55 [ INFO] * Parsing file: /home/ch/_sr/_sr3/data/objects/balloon1.mesh.xml ...
2023-11-26 17:07:56 [ INFO] - Done at 0.82 seconds
2023-11-26 17:07:56 [ INFO] * Collecting mesh data...
```
Error in Blender:
```
Python: Traceback (most recent call last):
  File "/home/ch/.config/blender/3.6/scripts/addons/io_ogre/ui/importer.py", line 184, in execute
    ogre_import.load(os.path.join(target_path, target_file_name))
  File "/home/ch/.config/blender/3.6/scripts/addons/io_ogre/ogre/ogre_import.py", line 1359, in load
    xCollectMeshData(meshData, xDocMeshData, onlyName, folder)
  File "/home/ch/.config/blender/3.6/scripts/addons/io_ogre/ogre/ogre_import.py", line 316, in xCollectMeshData
    sm['geometry'] = xCollectVertexData(subnodes)
  File "/home/ch/.config/blender/3.6/scripts/addons/io_ogre/ogre/ogre_import.py", line 263, in xCollectVertexData
    u = float(vt.getAttributeNode('u').value)
AttributeError: 'NoneType' object has no attribute 'value'
bpy.ops.ogre.import_mesh(filepath="/home/ch/_sr/_sr3/data/objects/balloon1.mesh.xml")
```

#### OgreXMLConverter.log or OgreMeshTool.log
```
Trying to read /home/ch/_sr/_sr3/data/objects/balloon1.mesh as a v1 mesh...
Failed.
Trying to read /home/ch/_sr/_sr3/data/objects/balloon1.mesh as a v2 mesh...
Success!
Saving as a v2 mesh...
MeshSerializer writing mesh data to stream ...
File header written.
Writing mesh data...
Writing submesh...
Submesh exported.
Exporting LOD level thresholds....
LOD level thresholds exported.
Exporting bounds information....
Bounds information exported.
Exporting submesh name table...
Submesh name table exported.
Exporting hash for caches...
Exporting hash for caches exported.
Mesh data exported.
MeshSerializer export successful.
```

**Comment by cryham** [comment]
Right, thanks. How can I get the command 'cmd' that gets executed printed out? If I do print in .py I don't see it, and those log messages in it don't appear in blender2ogre .log somehow.

**Comment by sercero** [comment]
Have you enabled DEBUG logging?
[blender2ogre Options](https://github.com/OGRECave/blender2ogre/blob/master/Options.md)

**Comment by rubberduck9999** [comment]
I was the one, who noticed import fails first, and now I figured out, that for me not just importing any mesh file using OgreMeshTools fails, but also modifies them when I try to import them. This leads to mesh files beeing corrupted in many cases.
[ogre_import_tests.zip](https://github.com/OGRECave/blender2ogre/files/13536503/ogre_import_tests.zip)

I wrote a bit more here in the stunt rally forum, what I did and figured out. 
https://forum.freegamedev.net/viewtopic.php?f=78&t=19742#p111745

**Comment by sercero** [comment]
So, you say that there is a problem with `OgreMeshTools` where the conversion from .mesh to .xml corrupts or modifies the original .mesh?

**Comment by rubberduck9999** [comment]
yes, at least for me, when I tried to import it with OgreMeshTools, files got unreadable, both in the game, but also for the old exporter, which only supports the v1 mesh version. Windows even showed a v2 mesh file that I tried to import set to v2 as modified, though this was the only case, where the file still worked in the game. Though in all cases with OgreMeshTools the import failed and files showed as modified.

I noticed that, when I then tried the old importer for importing at least a v1 version of that mesh, which always worked before I updated to OgreMeshTools, where I needed it the first time. First I thought that I may have configured something wrong, and tried a different file with the old importer, where it worked perfectly. That was the point where I saw, that the file that failed to import was modified, which I was able to reproduce again and again, testing if not just the import fails (on v1 and v2 mesh files with both importers), but also if stunt rally can load this file. In case the file was corrupted, SR editor crashed when I tried to select the object.

Also, I am currently not on my regular computer and will be back friday evening / saturday

**Comment by darksylinc** [comment]
This issue will be fixed by PR #256 

OgreMeshTool does not support exporting all types of attributes (e.g. half), and event if it did, blender2ogre would not know how to read some of this data (e.g. qtangents).

The PR fixes this because it passes `-U` as argument, which asks OgreMeshTool to convert the mesh into something standard, that blender2ogre can read (and also results in a valid XML file).
